### PR TITLE
mgmt/mcumgr/lib: Remove strnlen usage from OS cmd group and fix how  CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN is used

### PR DESCRIPTION
**Pull request by de-nordic**
The PR contains three commits that:
 1) fix Kconfig information on what `CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN` actually means, as previously it has been stated that it includes string terminating 0, which is not true as CBOR encoding does not include the 0 and the whole length could be used for thread name;
 2) Fixes echo command to not use strnlen making it C99 compliant;
 3) Fixes taskstat command to not use trnlen making it C99 compliant, fixes usage of `CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN` and reduces duplicated code.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c [hunk 1]
```diff
@@ -64,13 +65,14 @@ static inline CborError
...
+	const int max_size = MIN(ARRAY_SIZE(thread_name), ARRAY_SIZE(thread->name)) - 1;
 
 	ARG_UNUSED(idx);
+	memcpy(thread_name, thread->name, max_size);
+	thread_name[max_size] = '\0';
 
-	err = cbor_encode_text_string(encoder, thread->name, strnlen(thread->name,
-				      CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1));
-	return err;
+	return cbor_encode_text_stringz(encoder, thread->name);
```
Comment (tejlmand) [inline subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:75]: I guess you want to pass `thread_name` here ?

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c [hunk 2]
```diff
@@ -64,13 +65,14 @@ static inline CborError
 os_mgmt_taskstat_encode_thread_name(struct CborEncoder *encoder, int idx,
 				    const struct k_thread *thread)
 {
-	int err = 0;
+	char thread_name[CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN];
+	const int max_size = MIN(ARRAY_SIZE(thread_name), ARRAY_SIZE(thread->name)) - 1;
 
 	ARG_UNUSED(idx);
+	memcpy(thread_name, thread->name, max_size);
+	thread_name[max_size] = '\0';
```
Comment (tejlmand) [inline subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:73]: why not just do:
```
size_t name_length = strlen(thread->name);
if (name_length > CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1) {
    name_length = CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1;
}
err = cbor_encode_text_string(encoder, thread->name, name_length);
```

what's the reason for copying to a new array ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
I believe there is a flaw in code.

And then I have a minor proposal, which won't create an extra array.

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c (hunk 2 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c]: > 
> what's the reason for copying to a new array ?

Hm. Yes, that wasn't the smartest way to do this.

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c (hunk 1 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c]: Fixed.

**Review by nordicjm — DISMISSED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c [hunk 3]
```diff
@@ -65,50 +65,40 @@ static inline CborError
...
-{
-	CborError err = 0;
-	char thread_name[CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN];
+	size_t name_len = strlen(thread->name);
 
 	ARG_UNUSED(idx);
 
-	thread_name[sizeof(thread_name) - 1] = 0;
-	ll_to_s((int)thread->base.prio, sizeof(thread_name) - 1, thread_name);
+	if (name_len > (CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN)) {
```
Comment (tejlmand) [inline subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:72]: minor nit.

`CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN` is defined as an int, so you may remove the extra set of parenthesis.
https://github.com/zephyrproject-rtos/zephyr/blob/f2dde2819f8fed317ec7cc9432708db7609b0da6/subsys/mgmt/mcumgr/Kconfig#L274-L275

```suggestion
	if (name_len > CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN) {
```

**Review by tejlmand — DISMISSED** [review]
lgtm, a minor nit observed.

**Review by nordicjm — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm - re-approved

**Review by carlescufi — APPROVED** [review]

**Pull request merged**